analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp6.cs

```diff
@@ -62,109 +76,111 @@ public void TryCatch4(object o)
+            }
+        }
 
-    void NonCompliant2()
-    {
-      object o = null;
-      switch (o?.GetHashCode())
-      {
-        case null:
-          o.ToString(); // Noncompliant
-          break;
-        default:
-          break;
-      }
-    }
-  }
+        public void NonCompliant1(List<int> list, A a)
+        {
+            var row = list?.Count;
+            if (a.booleanVal = (row == null))
+            {
+                var type = list.ToArray(); // Noncompliant
+            }
+        }
 
-  public class ReproForIssue2338
-  {
-    public ConsoleColor Color { get; set; }
+        void Compliant2(object o)
+        {
+            switch (o?.GetHashCode())
+            {
+                case 1:
+                    o.ToString(); // Noncompliant FP, nullability is inferred from result relation
+                    break;
+                default:
+                    break;
+            }
+        }
 
-    public void Method1(ReproForIssue2338 obj)
-    {
-      switch (obj?.Color)
-      {
-        case null:
-          Console.ForegroundColor = obj.Color; // FIXME Non-compliant
-          break;
-        case ConsoleColor.Red:
-          Console.ForegroundColor = obj.Color; //compliant
-          break;
-        default:
-          Console.WriteLine($"Color {obj.Color} is not supported."); //compliant
-          break;
-      }
+        void NonCompliant2()
+        {
+            object o = null;
+            switch (o?.GetHashCode())
+            {
+                case null:
+                    o.ToString(); // Noncompliant
+                    break;
+                default:
+                    break;
+            }
+        }
     }
 
-    public void Method2(ReproForIssue2338 obj)
+    public class ReproForIssue2338
     {
-      obj = null;
-      switch (obj?.Color)
-      {
-        case ConsoleColor.Red:
-          Console.ForegroundColor = obj.Color;  // FIXME was compliant before
-//                                  ^^^    {{'obj' is null on at least one execution path.}}
-          break;
-        default:
-          Console.WriteLine($"Color {obj.Color} is not supported."); // Noncompliant
-//                                   ^^^    {{'obj' is null on at least one execution path.}}
-          break;
-      }
+        public ConsoleColor Color { get; set; }
+
+        public void Method1(ReproForIssue2338 obj)
+        {
+            switch (obj?.Color)
+            {
+                case null:
+                    Console.ForegroundColor = obj.Color; // Noncompliant
+                    break;
+                case ConsoleColor.Red:
+                    Console.ForegroundColor = obj.Color; // Noncompliant FP, requires nullability support in captured flow operation
+                    break;
+                default:
+                    Console.WriteLine($"Color {obj.Color} is not supported."); // Noncompliant FP, requires nullability support in captured flow operation
```

Comment:
This one is tricky.

Reply:
Oh yes. I took some time to think about it. If we can already support it. But no, it explodes too much.